scripts/ultracool_sheet/generate_simple_links.py

```diff
@@ -0,0 +1,80 @@
+from scripts.ingests.utils import load_simpledb, find_source_in_db
+from astropy.io import ascii
+from urllib.parse import quote
+import requests
+from astropy.table import Table
+
+
+RECREATE_DB = False
+db = load_simpledb("SIMPLE.sqlite", recreatedb=RECREATE_DB)
+
+# Load Ultracool sheet
+sheet_id = "1i98ft8g5mzPp2DNno0kcz4B9nzMxdpyz5UquAVhz-U8"
+# link = f"https://docs.google.com/spreadsheets/d/{sheet_id}/export?format=csv"
+link = "scripts/ultracool_sheet/UltracoolSheet - Main_010824.csv"
+
+# read the csv data into an astropy table
+uc_sheet_table = ascii.read(
+    link,
+    format="csv",
+    data_start=1,
+    header_start=0,
+    guess=False,
+    fast_reader=False,
+    delimiter=",",
+)
+
+# Match sources in Ultracool sheet to sources in SIMPLE
+uc_names = []
+simple_urls = []
+simple_sources = []
+for source in uc_sheet_table[1647:]:
+    uc_sheet_name = source["name"]
+    match = find_source_in_db(
+        db,
+        uc_sheet_name,
+        ra=source["ra_j2000_formula"],
+        dec=source["dec_j2000_formula"],
+    )
+
+    # convert SIMPLE source name to URL
+    if len(match) == 0:
+        msg = f"No match found for {uc_sheet_name}"
+        raise ValueError(msg)
+    elif len(match) > 1:
+        msg = f"Multiple matches found for {uc_sheet_name}"
+        raise ValueError(msg)
+    elif len(match) == 1:
+        simple_source = match[0]
+        print(f"Match found for {uc_sheet_name}: {simple_source}")
+    else:
+        raise ValueError("Unexpected state")
+
+    # URLify source name
+    source_url = quote(simple_source)
+    url = "https://simple-bd-archive.org/solo_result/" + source_url
+
+    # test the URL to make sure it is valid
+    url_status = requests.head(url).status_code
+    if url_status != 200:
+        raise ValueError("URL not valid for ", uc_sheet_name, simple_source, url)
+    else:
+        print("URL valid for ", uc_sheet_name, simple_source, url)
```

Comment:
```suggestion
    if url_status == 404:
        raise ValueError("URL not valid for ", uc_sheet_name, simple_source, url)
    elif url_status != 200:
        print("URL valid for ", uc_sheet_name, simple_source, url, " BUT the webpage is having issue: ", url_status)
    else:
        print("URL valid for ", uc_sheet_name, simple_source, url)
```

Reply:
Let's do something like give a warning if it's a timeout.